pkg/catalogv2/helmop/operation.go

```diff
@@ -1097,3 +1130,22 @@ func (s *Operations) createPod(secretData map[string][]byte, kustomize bool, ima
 		ImageOverride: imageOverride,
 	}
 }
+
+// addCpTaintsToTolerations gets the list of control plane nodes and adds their taints to the given tolerations.
+func (s *Operations) addCpTaintsToTolerations(tolerations []corev1.Toleration) ([]corev1.Toleration, error) {
```

Comment:
The logic is almost the same. In the code you sent, it lists all nodes, then filters by CP labels and then gets the taints. In this PR, it lists all nodes already filtered by label, then return a toleration for each taint. 
I thought it would be easier to understand this way.